addon/resolver.js

```diff
@@ -0,0 +1,55 @@
+const Resolver = require('ember-resolver')['default']
+
+Resolver.reopen({
+  expandLocalLookup: function (targetFullName, sourceFullName) { // eslint-disable-line complexity
+    const modulePrefix = this.namespace.modulePrefix
+    const podModulePrefix = this.namespace.podModulePrefix
+    const parsedTarget = this.parseName(targetFullName)
+    const parsedSource = this.parseName(sourceFullName)
+
+    let sourceName = parsedSource.fullNameWithoutType
+    let targetName = parsedTarget.fullNameWithoutType
+    const targetType = parsedTarget.type
+
+    // Local lookup only applies to targets referenced from a template
+    if (parsedSource.type !== 'template') {
+      return `${parsedTarget.type}:${sourceName}/${targetName}`
```

Comment:
This is one portion of the original code I was curious about - I never see this condition met in any of the current tests.  Is this a necessary check or can it be removed?